### PR TITLE
Add shortcuts and actions for default focus traversal

### LINKED ISSUE #13264: Users should be able to navigate around the app using keyboard
None of the navigational shortcuts work. I should be able to navigate Flutter apps installed on ChromeOS using arrows/tab/enter key.s The only key that works is Esc which quits the app.

### LINKED ISSUE #31957: Game controller and DPAD navigation should work
Internal: b/140829915

Along with keyboard navigation (#13264), and handling keycodes for DPAD (#1670), we should also support focus navigation with a game controller and phone DPAD.

### PR DESCRIPTION
**Pull request by gspencergoog**
## Description

This adds the default shortcuts and actions for keyboard-based focus traversal of apps. 

This list of shortcuts includes shortcuts for TAB, SHIFT TAB, RIGHT_ARROW, LEFT_ARROW, UP_ARROW, DOWN_ARROW, and the four DPAD keys for game controllers (because the DPAD produces arrow key events).

It doesn't yet include functionality for triggering a control (e.g. SPACE, ENTER, or controller buttons), because that involves restructuring some of the Flutter controls to trigger animations differently, and so will be done in another PR (https://github.com/flutter/flutter/pull/41220)

## Related Issues

Fixes #13264
Fixes #31957

## Tests

- Added tests to make sure that navigation works by simulating keystrokes. (not yet complete)

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I signed the [CLA].
- [X] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] All existing and new tests are passing.
- [X] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [X] I am willing to follow-up on review comments in a timely manner.
- [X] I checked all the boxes!

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [X] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

**Comment by gspencergoog** [comment]
This needs tests, so I'm not going to move it out of "draft" until I'm done with those.

File: packages/flutter/test/widgets/focus_traversal_test.dart [hunk 1]
```diff
@@ -849,5 +851,112 @@ void main() {
...
+            );
+          },
+        ),
+      );
+
+      // Initial focus happens.
+      expect(Focus.of(upperLeftKey.currentContext).hasPrimaryFocus, isTrue);
+      await tester.sendKeyEvent(LogicalKeyboardKey.tab);
+      expect(Focus.of(upperRightKey.currentContext).hasPrimaryFocus, isTrue);
+      // Initial focus happens.
```
Comment (darrenaustin) [inline packages/flutter/test/widgets/focus_traversal_test.dart]: Is this a copy/paste oversight? How is the initial focus happening again?

**Review by darrenaustin — APPROVED** [review]
LGTM.

File: packages/flutter/test/widgets/focus_traversal_test.dart (hunk 1 above)
Comment (gspencergoog) [reply packages/flutter/test/widgets/focus_traversal_test.dart]: Yeah, for some reason I CAN'T READ.  Removed all of the copies of that, and fixed up the first one (which was the only place that comment belonged).

**Pull request merged**